### DISCUSSION TITLE
i3 and KDE notifications

### DISCUSSION THREAD

**Discussion by FikaMedHasse** [i3/i3#5417] [Q&A]
I am using KDE Plasma 5.26 with i3 instead of kwin. For the most part it works wonderful, but KDE notifications appear in the top-left corner of the screen, instead of the supposed bottom-right. They also apperar part-way off the screen, so it's not an issue with KDE, more likely with where i3 chooses to place the notification. Attached are the relevant lines from my i3 config:

```
# don't treat plasma pop-ups as full-sized windows
for_window [class="plasmashell"] floating enable

# don't spawn empty window for plasma desktop
for_window [title="Desktop — Plasma"] kill; floating enable; border none

# dont steal focus
no_focus [class="plasmashell" window_type="notification"]
no_focus [class="plasmashell" window_type="on_screen_display"]
for_window [window_role="pop-up"] floating enable
for_window [window_role="task_dialog"] floating enable
```
Any idea how I can make the notifications appear in the right place?

**Comment by Airblader** [comment]
Getting xprop and xwininfo of the notification would be a first step. Most likely they are override_redirect, meaning the window manager doesn't touch them and i3 is out of the question.

If they are managed windows, it's entirely possible KDE uses some proprietary protocol that i3 doesn't implement, they tend to do that.

Knowing the geometry the window requests (i3 logs) would give you an answer on what i3 knows about and how it decides to place it (if it's managed).

**Reply by FikaMedHasse** [reply]
I don't see any overwrite_redirect in those outputs. here they are from KRunner, which behaves the same way as the notifications but is easier to open on demand: 
[xwininfooutput.txt](https://github.com/i3/i3/files/10713147/xwininfooutput.txt)
[xpropoutput.txt](https://github.com/i3/i3/files/10713148/xpropoutput.txt)

**Reply by Airblader** [reply]
> Override Redirect State: no

At this point I'd look into the i3 logs to see how i3 decides to place it.

**Reply by FikaMedHasse** [reply]
I tried looking at the i3 docs to find where the logs are, but it seems like I have to modify the xsession? it looks like this now: 
[plasma-i3.txt](https://github.com/i3/i3/files/10713453/plasma-i3.txt)
(i renamed it from .desktop to .txt because github didn't like the filetype)

how do i enable logs? i tried appending `>/home/[USER]/i3log-$(date +'%F-%k-%M-%S') 2>&1` to the `exec /usr/bin/i3` but then it wouldn't start at all. Does i3 not keep regular log files?

**Reply by FikaMedHasse** [reply]
I did some more research. It seems like I WANT to have the override_redirect set, so that i3 does not touch them and lets them popup whereever. Is there an easy way to set that flag for all relevant windows (notifications and KRunner)?

**Reply by Airblader** [reply]
override_redirect is set by the application. If it doesn't set it, you don't want it to be set.

(FYI, I'm not near my laptop to look at the logs)

**Comment by mksonkar** [comment]
You can change the notification position in KDE Settings > Notifications section.
